components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -305,7 +310,21 @@ public OAuth2AccessTokenRespDTO issue(OAuth2AccessTokenReqDTO tokenReqDTO)
 
         if (tokReqMsgCtx.getScope() != null && OAuth2Util.isOIDCAuthzRequest(tokReqMsgCtx.getScope())) {
             IDTokenBuilder builder = OAuthServerConfiguration.getInstance().getOpenIDConnectIDTokenBuilder();
-            tokenRespDTO.setIDToken(builder.buildIDToken(tokReqMsgCtx, tokenRespDTO));
+            String idToken = builder.buildIDToken(tokReqMsgCtx, tokenRespDTO);
+            try {
+                JWT jwt = JWTParser.parse(idToken);
```

Comment:
How about using the exception handling capabilities for this.
This method has a throws clause for IdentityOAuth2Exception.
 
So, how abt creating a sub-class of IdentityOAuth2Exception say, IDTokenValidationFailureException? It gets thrown when the validation fails. 
We can catch and handle this.

We need to special-case the default ID Token builder to overcome possible penalty on performance of building the id token twice.